test/components/geojson_test.dart

```diff
@@ -171,39 +204,268 @@ main() {
+                ),
+              ],
+            ),
+            id: 1,
+            properties: {"key": "val"}),
+      ],
+    );
+    final cloned = coll.clone();
+    final feat = cloned.features.first;
+    final bbox = BBox(100, 0, 101, 1);
+    expect(cloned.bbox, bbox);
+    expect(feat.id, 1);
+    expect(feat.bbox, bbox);
+    expect(feat.properties!.keys.first, "key");
+    expect(feat.properties!.values.first, "val");
+    expect(feat.geometry!, isA<GeometryCollection>());
+    final geomColl = feat.geometry!;
+    expect(geomColl.geometries.length,
+        coll.features.first.geometry!.geometries.length);
+    for (var geom in geomColl.geometries) {
+      expect(geom.bbox, isNotNull);
+      expect(geom.coordinates, isNotEmpty);
+
+      _expandRecursively(List inner) {
+        if (inner is List<Position>) {
+          return inner;
+        } else {
+          return inner.expand((el) => el is List ? _expandRecursively(el) : el);
+        }
+      }
+
+      var expanded = _expandRecursively(geom.coordinates);
+      expect(
+        expanded.first,
+        Position(100, 0),
+      );
+    }
+    // TODO refine tests
+  });
+
+  final points = [
+    Point(coordinates: Position(1, 2, 3)),
+    Point(coordinates: Position(2, 1, 3)),
+    Point(coordinates: Position(3, 2, 1)),
+  ];
+  test('MultiPoint.fromPoints', () {
+    var a = MultiPoint.fromPoints(points: points);
+    expect(a.coordinates.first, Position(1, 2, 3));
+    expect(() => MultiPoint.fromPoints(points: []),
+        throwsA(isA<AssertionError>()));
+  });
+  test('LineString.fromPoints', () {
+    var a = LineString.fromPoints(points: points);
+    expect(a.coordinates.first, Position(1, 2, 3));
+    expect(() => LineString.fromPoints(points: []),
+        throwsA(isA<AssertionError>()));
+  });
+  test('MultiLineString.fromLineStrings', () {
+    var a = MultiLineString.fromLineStrings(lineStrings: [
+      LineString.fromPoints(points: points),
+      LineString.fromPoints(points: points)
+    ]);
+    expect(a.coordinates.first.first, Position(1, 2, 3));
+    expect(() => MultiLineString.fromLineStrings(lineStrings: []),
+        throwsA(isA<AssertionError>()));
+  });
+  test('Polygon.fromPoints', () {
+    var a = Polygon.fromPoints(points: [points]);
+    expect(a.coordinates.first.first, Position(1, 2, 3));
+    expect(
+        () => Polygon.fromPoints(points: []), throwsA(isA<AssertionError>()));
+  });
+  test('MultiPolygon.fromPolygons', () {
+    var a = MultiPolygon.fromPolygons(polygons: [
+      Polygon.fromPoints(points: [points]),
+      Polygon.fromPoints(points: [points])
+    ]);
+    expect(a.coordinates.first.first.first, Position(1, 2, 3));
+    expect(() => MultiPolygon.fromPolygons(polygons: []),
+        throwsA(isA<AssertionError>()));
+  });
+
+// examples
+// copied from RFC 7946 https://datatracker.ietf.org/doc/html/rfc7946
+// copied from Wikipedia https://en.wikipedia.org/wiki/GeoJSON#Geometries
+  group('Example file', () {
```

Comment:
perhaps just double check the licensing for these code examples to ensure they are compliant with this project's outbound license

Reply:
Wikipedia seems to be Creative Commons Attribution-ShareAlike License

RFC examples:
```
Code Components extracted from this document must include Simplified BSD License ...
```

Reply:
added license notices